elfi/model/extensions.py

```diff
@@ -50,23 +59,28 @@ def pdf(this, x, *params, **kwargs):
         """
         raise NotImplementedError
 
+    @classmethod
     def logpdf(this, x, *params, **kwargs):
         """Log of the probability density function at x.
 
         Parameters
         ----------
         x : array_like
-            points where to evaluate the pdf
+            points where to evaluate the logpdf
         param1, param2, ... : array_like
             parameters of the model
         kwargs
 
         Returns
         -------
-        pdf : ndarray
+        logpdf : ndarray
            Log of the probability density function evaluated at x
         """
-        raise NotImplementedError
+        p = this.pdf(x, *params, **kwargs)
+        if p == 0:
+            return -np.inf
```

Comment:
I think this is not going to work with vectorized arguments. Also I think `np.log` returns -inf for zeros automatically. Some distributions may be more conveniently written directly as logpdf and then pdf is taken as the exponent.